src/map/pc.cpp

```diff
@@ -14989,6 +14994,364 @@ void pc_attendance_claim_reward( struct map_session_data* sd ){
 	clif_attendence_response( sd, attendance_counter );
 }
 
+/**
+ * Save a captcha image to memory via /macro_register.
+ * @param sd: Player data
+ * @param image_size: Captcha image size
+ * @param captcha_answer: Answer to captcha
+ */
+void pc_macro_captcha_register(map_session_data &sd, int16 image_size, char captcha_answer[CAPTCHA_ANSWER_SIZE]) {
+	nullpo_retv(captcha_answer);
+
+	sd.captcha_upload.cd = nullptr;
+	sd.captcha_upload.upload_size = 0;
+
+	if (strlen(captcha_answer) < 4 || image_size < 0 || image_size > CAPTCHA_BMP_SIZE || captcha_db.size() >= 0xFFF) {
```

Comment:
```suggestion
	if (strlen(captcha_answer) < 4 || image_size < 0 || image_size > CAPTCHA_BMP_SIZE) {
```
Still thinking if we should remove all this. It is actually useless.
We only limit ourselves with this captcha_key.